extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/VertxWebProcessor.java

```diff
@@ -91,8 +92,12 @@ FeatureBuildItem feature() {
     }
 
     @BuildStep
-    public void kubernetes(HttpConfiguration config, BuildProducer<KubernetesPortBuildItem> portProducer) {
-        portProducer.produce(new KubernetesPortBuildItem(config.port, "http"));
+    public void kubernetes(BuildProducer<KubernetesPortBuildItem> portProducer) {
+        // we can't use HttpConfiguration since the port is a runtime configuration value
+        // therefore we just read the value from MP Config - this is fine since the value is only ever used to create
+        // Kubernetes resources at build time
+        portProducer.produce(new KubernetesPortBuildItem(
+                ConfigProvider.getConfig().getOptionalValue("quarkus.http.port", Integer.class).orElse(8080), "http"));
```

Comment:
There's no guarantee that the value will be present at build time.

Reply:
Doesn't the use of `Optional` cover that? Am I missing something?

Reply:
If I built without `quarkus.http.port` and then run with it set to 9999, then it will fail.  If I know that I need one value for build and another for run then maybe it's better to have a separate property that can be set.  You could also add in a run time check that prints a log message if the ports don't match up, if you wanted to.

Reply:
That is certainly true, however it won't exactly be a failure - basically the Kubernetes resource file will not match what is being used at runtime, but there is no way to match that anyway. 

I would assume that users already know this since they know that there Kubernetes files are being generated at build time.

Adding a kubernetes port build item (and configuration to go with it) won't really get users much IMHO, since it won't be able to change anything at runtime either - meaning they are in the same spot as if they didn't have it.

Am I missing something?

Reply:
Absolutely fair point. 
However wouldn't the same logic apply if there were 2 config properties, that need to change in tandem (and one only being configurable at build time)? Wouldn't the user need to infer that `quarkus.http.port` and `quarkus.kubernetes.port` need to point  to the same value?
Now of course like you said you could have a runtime check (and probably a build time check as well), but I am not convinced it's a better experience for users.

I'll leave it up to you however, if you think it should be 2 different config items, then I'll do that :)

Reply:
Another thing if we add 2 different config items, is what would be the name of the config? Asking because `quarkus.kubernetes` is already being used by `KubernetesConfig` in the `quarkus-kubernetes` extension

Reply:
It just seems weird to add a kubernetes port property (which is exactly the same as the regular http port) in the vertx configuration.
I know that I myself as a user would question why I have to do:

```
quarkus.http.port=9090
quarkus.http.kubernetes-port=9090
```

instead of just 

```
quarkus.http.port=9090
```

which would cover both cases. And of course it could be documented that changing the port at runtime would have no effect on the generated configuration.

Furthemore I think users will really not like having the port property being a build time only property. I personally don't mind, but I think a lot of users would.

So, do you want me to go ahead with the 2 properties?

Reply:
Just thought of another solution which I think would give the best of both worlds:

We keep the single port, but at build time we also generate a class that is run at application startup and checks whether the runtime port is the different than the build time one and if so log a warning about the generates Kubernetes resources not being in sync with the runtime config.

WDYT @dmlloyd ?

Reply:
OK, I'll proceed to code it then

Reply:
Need to step out for a few hours. I'll continue when and push the updates to the PR when I get back

Reply:
@dmlloyd PR updated